src/contracts/core/DelegationManager.sol

```diff
@@ -480,6 +494,23 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         return approverDigestHash;
     }
 
+    /** 
+     * @notice Returns 'true' if the staker can undelegate or  if the staker is already undelegated, and 'false' otherwise
+     * @dev A staker can only undelegate if they have no "active" shares in EigenLayer and are not themselves an operator
+     */
+    function stakerCanUndelegate(address staker) public view returns (bool) {
```

Comment:
Outside the scope of this PR, the below naming is more straightforward.
```
function canUndelegate(address staker) public view returns (bool)
```
The notice could also be shortened to:
```
@notice Checks if `staker` can undelegate.
```

Reply:
I like this suggestion, will plan to implement

Reply:
addressed in minor commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/1124853d65e922c18369cc94a84dd806ffea7ac1